### DISCUSSION TITLE
How to load a custom module

### DISCUSSION THREAD

**Discussion by Ncturn** [directus/directus#4031] [Archive]
Hi, im new in directus and now im creating a new module with this guide:
https://docs.directus.io/guides/modules/
i used the default examples to create the index.js and i paste it in the extension folder but it does not load and i cant get access to the custom module, anyone can helpme or guide me to create a custom module.

Thanks¡¡.

**Comment by rijkvanzanten** [comment]
> i used the default examples to create the index.js and i paste it in the extension folder

Did you follow the last where it's bundled and copied that output? https://docs.directus.io/guides/modules/#_4-build-and-deploy


Hard to tell what's going on here without more examples of what you tried exactly 🙂 What does your extensions folder look like currently?

**Reply by Ncturn** [reply]
Hi, sorry this are my files:
index.js:
import  ModuleComponent from './module.vue';

export default {
	id: 'banner',
	name: 'Banners',
	icon: 'box',
	routes: [
		{
			path: '/',
			component: ModuleComponent,
		},
	],
};
module.vue:
<template>
	<private-view title="My Custom Module">Content goes here...</private-view>
</template>

<script>
export default {};
</script>


and i used the command npx rollup -c with this config:

import { terser } from 'rollup-plugin-terser';
import resolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import vue from 'rollup-plugin-vue';

export default {
	input: 'src/index.js',
	output: {
		format: 'es',
		file: 'dist/index.js',
	},
	plugins: [terser(), resolve(), commonjs(), vue()],
};

the builded index.js created i paste it in extension/modules i run directus but i cant see the module:
![image](https://user-images.githubusercontent.com/43647373/107803605-6ecd4d80-6d28-11eb-9948-306aace33b84.png)

thanks for your help.

**Reply by rijkvanzanten** [reply]
Ah I see. You gotta put the whole `dist` folder under `modules` (you can rename it to something else if you want), not just the `index.js` file 🙂

**Reply by Ncturn** [reply]
@rijkvanzanten thanks a lot it works¡¡

**Reply by ashx404** [reply]
I'm trying to do the same, but for some reason the path of index.js which is getting generated is incorrect. 
![image](https://user-images.githubusercontent.com/45964517/125512849-4a81be27-ccbb-4bca-a17f-41e43d30f452.png)

while it should be this
file:///D:/starCast/directus/proj/extensions/modules/calender/index.js

Whenever I hit that route from my admin, I can see a request but due to wrong path, I am getting Page Not Found error :(
Please help.

**Comment by faisal1357** [comment]
![Capture](https://user-images.githubusercontent.com/45657326/148962974-8d3d8eb0-facc-445f-82b4-1c68b25b8b9a.JPG)
after build i can't see my module

**Reply by rijkvanzanten** [reply]
@faisal1357 Make sure that only the dist/index.js file is included in your sub-user-setting folder (so it should be `/extensions/modules/sub-user-setting/index.js`)

**Reply by faisal1357** [reply]
![image](https://user-images.githubusercontent.com/45657326/148966371-ecc1c939-3908-4801-b9c7-4b992a331dd1.png)
index.js file is included in dist file

**Reply by nickrum** [reply]
@faisal1357 The path of the root component of your module has to be the empty string (`path: ''` instead of `path: '/'`).
